geometry/convex_hull.cpp

```diff
@@ -0,0 +1,69 @@
+#include "geometry/convex_hull.hpp"
+
+#include "geometry/robust_orientation.hpp"
+
+#include "base/stl_helpers.hpp"
+
+#include <algorithm>
+
+using namespace std;
+
+namespace m2
+{
+namespace
+{
+double const kEps = 1e-9;
+
+// Checks whether (p1 - p) x (p2 - p) > 0.
+bool IsCCW(PointD const & p1, PointD const & p2, PointD const & p)
+{
+  return robust::OrientedS(p1, p2, p) > kEps;
+}
+
+bool ContinuesHull(vector<PointD> const & hull, PointD const & p)
+{
+  auto const n = hull.size();
+  if (n < 2)
+    return true;
+
+  auto const & p1 = hull[n - 2];
+  auto const & p2 = hull[n - 1];
+
+  // Checks whether (p2 - p1) x (p - p2) > 0.
+  return robust::OrientedS(p1, p, p2) < -kEps;
```

Comment:
I suggest you reuse IsCCW here and remove the comment.

Reply:
How to reuse IsCCW here?

Reply:
```
    (p2 - p1) x (p - p2) > 0
≡ { anticommutativity of cross product }
    - (p - p2) x (p2 - p1) > 0
≡ { s(a x b) = (a x sb) for scalar s }
    (p - p2) x (p1 - p2) > 0
≡
    IsCCW(p, p1, p2)
```

Reply:
Thanks, done!